### ISSUE TITLE
LSP request failure toasts cannot be turned off

### ISSUE THREAD

**Issue by dibarbet** [microsoft/vscode-languageserver-node#1449]
Not sure if this is an issue here or on the vscode side (or even a bug at all) - but toast notifications due to a failed LSP request cannot be turned off.

To repro
1.  Clone and open the vscode extension samples LSP sample - https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-sample
2.  add a line to throw an error in the completion request
3.  run the extension and trigger completion
4.  note the failure toast, and attempt to turn off the notifications for the extension
<img width="351" alt="image" src="https://github.com/microsoft/vscode-languageserver-node/assets/5749229/18eb5fdf-7e9c-40bf-872a-8eeef3673d7d">
5.  dismiss the failure, re-invoke completion and note that the toast continues to show up.

Was reported in https://github.com/dotnet/vscode-csharp/issues/6977
We're attempting to fix the issues causing the request to fail, but it seems like a bug that the notifications cannot be turned off.

**Comment by dbaeumer** [comment]
@bpasero is there anything special I need to do. When I press the Manage Extension button I get this:


![Image](https://github.com/microsoft/vscode-languageserver-node/assets/1931590/99c12279-19fe-4b85-a9d9-28fb6459c2a5)

**Comment by bpasero** [comment]
@dbaeumer errors are excempted from notification ignore. Only info and warning can be ignored.

**Comment by dibarbet** [comment]
> errors are excempted from notification ignore.

I thought it might be something like that.  If that is the behavior, being able to configure the notification ignore from an error notification can be a little misleading.

**Comment by bpasero** [comment]
Yeah I do understand, will think about it. I am not a big fan of just not showing this action based on severity because it may still be hard for a user to understand what conditions there are to hide it.

**Comment by dbaeumer** [comment]
I will close the issue then.

**Comment by garrettlondon1** [comment]
@bpasero @dbaeumer Thank you for your consideration of this issue.

Please review https://github.com/dotnet/vscode-csharp/issues/6977 , as this is an extension which uncontrollably throws errors.

From a Accessibility perspective, not giving users the ability to disable errors, or extension authors ability, leaves VS code in a state where errors are coming up all the time, **even in Zen Mode**

Would you please consider reopening this ticket, as it is a huge accessibility pain and am unable to work properly with the C# extension in VS Code.

**Comment by dbaeumer** [comment]
IMO hiding errors from the user is not the right thing to do, since they usually shouldn't happen frequently


@dibarbet if this happens often with the CSharp server you might want to think about handling them special in the middleware of override `handleFailedRequest` on the client side. I would even considering add something to the client option to have a handler for them. Would this help?

**Comment by dbaeumer** [comment]
You should be able to do that in the middleware. If not ping me.
